### PR TITLE
Update Bitarray to Version 2.3.4

### PR DESCRIPTION
**Pull request by varlackc**
`bitarray` version `2.3.4`
1. check the upstream
    https://github.com/ilanschnell/bitarray/tree/2.3.4

2. check the pinnings

https://github.com/ilanschnell/bitarray/blob/2.3.4/update_doc.py

https://github.com/ilanschnell/bitarray/blob/2.3.4/setup.py

3. check the changelogs
    https://github.com/ilanschnell/bitarray/blob/2.3.4/CHANGE_LOG
    
    Ther were only bug fixes and feature additions in the changelog. 

    There were no breaking changes mentioned

4. additional research
    https://github.com/conda-forge/bitarray-feedstock/issues

    There were no breaking changes mentioned in conda-forge

5. verify dev_url
    https://github.com/ilanschnell/bitarray

6. verify doc_url
    https://pypi.python.org/pypi/bitarray/0.8.1

7. pip in the test section
8. veriy the test section
9. additional tests

In order to test the `bitarray` package version `2.3.4` the folowing
command was used:
`conda build bitarray-feedstock --test`
the test result was the following:
`All tests passed`

Based on the research findings and the test results we can conclude
that it is safe to update `bitarray` to version `2.3.4`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 21f2e10420f9ae74d3bdfd3cb74947ddbe9fbc0b2e5662f2f039001954f1d8b6
+  sha256: f19c62425576d3d1821ed711b94d1a4e5ede8f05ca121e99b6d978ed49c7a765
 
 build:
   number: 1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:13]: We should reset build number to `0`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,12 +26,13 @@ requirements:
...
+    - python >=3.6
 
 test:
   imports:
     - bitarray
   requires:
+    - python <3.10
     - pip
   commands:
     - pip check
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: The license should be `license: PSF-2.0`, see https://spdx.org/licenses/PSF-2.0.html

**Comment by skupr-anaconda** [comment]
Also, we need to build on s390x but it will fail due to DependenciesConflict

**Comment by skupr-anaconda** [comment]
Tests failed on `s390x`:
```
FAILED (failures=24, errors=1)

Traceback (most recent call last):

  File "/opt/conda/conda-bld/bitarray_1639775601698/test_tmp/run_test.py", line 7, in <module>

    assert bitarray.test().wasSuccessful()

AssertionError

Tests failed for bitarray-2.3.4-py37h2a837d6_1.tar.bz2 - moving package to /opt/conda/conda-bld/broken
```
@katietz How can we fix it? I have no idea yet...

**Comment by ilanschnell** [comment]
@varlackc @katietz @skupr-anaconda
There was a bug in bitarray itself which only reveals itself on bit-endian machines (https://github.com/ilanschnell/bitarray/pull/160).  I was able to fix the upstream problem and made a new 2.3.5 release today.

**Comment by skupr-anaconda** [comment]
@varlackc This PR can be closed. Here is a new PR of v2.3.5 https://github.com/AnacondaRecipes/bitarray-feedstock/pull/13 where the s390x issue was fixed

**Pull request closed without merging**